system/services/admin/LoginService.cfc

```diff
@@ -677,20 +678,23 @@ component displayName="Admin login service" {
 				two_step_auth_key_in_use = true
 			} );
 
+			var authMethodCookieValue = CreateUUID();
+			var loginRecordData       = { logged_in_date=Now(), cookie=authMethodCookieValue };
+
+			_getCookieService().setVar( name=_getTwoFactorAuthCookiePersistKey(), value=authMethodCookieValue );
+
 			var loginRecordDao = $getPresideObject( "security_user_two_factor_login_record" );
-			var updated = loginRecordDao.updateData( filter={
-				  security_user = userId
-				, ip_address    = arguments.ipAddress
-				, user_agent    = arguments.userAgent
-			}, data={ logged_in_date=Now() } );
+			var updated        = loginRecordDao.updateData(
+				  data   = loginRecordData
+				, filter = { security_user=userId, user_agent=arguments.userAgent }
+			);
 
 			if ( !updated ) {
-				loginRecordDao.insertData({
-					  security_user  = userId
-					, ip_address     = arguments.ipAddress
-					, user_agent     = arguments.userAgent
-					, logged_in_date = Now()
-				});
+				loginRecordData.security_user = userId;
+				loginRecordData.user_agent    = arguments.userAgent;
```

Comment:
Remove the user agent and IP here if removing from the object (see prior feedback)